contracts/RewardManager.sol

```diff
@@ -145,31 +160,58 @@ contract RewardManager is Ownable, Versionable, Safe {
     onlyHandlers
     returns (address)
   {
-    // creation of reward safe
-    // - enable person to claim rewards
-    address rewardSafe = rewardSafes[rewardProgramID][prepaidCardOwner];
-    require(
-      rewardSafe == ZERO_ADDRESS,
-      "prepaid card owner already registered for reward program"
-    );
     address[] memory owners = new address[](2);
-
     owners[0] = address(this);
     owners[1] = prepaidCardOwner;
-    rewardSafe = createSafe(owners[1]);
-    rewardSafes[rewardProgramID][prepaidCardOwner] = rewardSafe;
+    uint256 salt = _createSalt(rewardProgramID, prepaidCardOwner);
+    address rewardSafe = create2Safe(owners, 2, salt);
+    rewardSafes[rewardProgramID].add(rewardSafe);
     emit RewardeeRegistered(rewardProgramID, prepaidCardOwner, rewardSafe);
     return rewardSafe;
   }
 
-  //Public View Functions
+  function transferRewardSafe(
+    address newOwner,
+    uint256 safeTxGas,
+    uint256 baseGas,
+    uint256 gasPrice,
+    address gasToken,
+    bytes calldata signature
```

Comment:
> and to be clear, is this the signature you were mentioning that was generated offchain and is the abi encoded params for the gnosis safe exec call to do the transfer?

Actually, there are two gnosis transactions for `transferRewardSafe` and each of their signatures are generated offchain. When I said the signatures where generated offchain, it is meant to highlight the difference between your implementation and mine. You tend to form the whole signature (eoa sign+ contract sign + verifying data) from within the contract itself but I couldn't follow this because you use the relayer to initiate the transaction but I need to initiate the transaction from a safe.